### PR TITLE
feature: Add Open Booking API and CSP template support

### LINKED ISSUE #14: Add documentation for accessService
Add the following to the documentation example:

```php
$renderer = new TemplateRenderer();
        $baseUrl = $this->router->generate('open_active_dataset', ['venueId' => $venueId], RouterInterface::ABSOLUTE_URL) . '/';
        $baseOrdersUrl = str_replace('/orders-rpde', '', $this->router->generate('api_open_active_orders_get_collection', [], RouterInterface::ABSOLUTE_URL) . '/');
        return new Response($renderer->renderSimpleDatasetSite([
            'openDataFeedBaseUrl' => $baseUrl,
            'datasetSiteUrl' => $baseUrl,
            'datasetDiscussionUrl' => 'https://github.com/playfinder/openactive/issues',
            'datasetDocumentationUrl' => 'https://developer.openactive.io/',
            'datasetLanguages' => ['en-GB'],
            'organisationName' => $venue->getName(),
            'organisationUrl' => 'https://www.bookteq.com/',
            'organisationLegalEntity' => $venue->getName(),
            'organisationPlainTextDescription' => '',
            'organisationEmail' => $venue->getContactEmail(),
            'organisationLogoUrl' => 'https://s3.eu-west-2.amazonaws.com/cdn.bookteq.com/logo-small.png',
            'platformName' => 'Bookteq',
            'platformUrl' => 'https://www.bookteq.com/',
            'platformSoftwareVersion' => '1.0',
            'backgroundImageUrl' => 'https://s3.eu-west-2.amazonaws.com/cdn.bookteq.com/background-bookteq.png',
            'dateFirstPublished' => '2020-07-15',
            'accessService' => new \OpenActive\Models\OA\WebAPI([
                    '@type' => 'WebAPI',
                    'conformsTo' => [
                        'https://openactive.io/open-booking-api/1.0/#core'
                    ],
                    'endpointDescription' => 'https://openactive.io/open-booking-api/1.0/swagger.json',
                    'endpointURL' => $baseOrdersUrl,
                    'landingPage' => $baseUrl,
                    'name' => $venue->getName(),
                    'documentation' => 'https://developer.openactive.io/'
                ])
        ], self::FEED_TYPES), 200, [
            'Content-Type' => 'text/html'
        ]);
```

### PR DESCRIPTION
**Pull request by nickevansuk**
Add native Open Booking API support (fixes #14), in addition to [CSP support](https://developer.openactive.io/publishing-data/dataset-sites#option-2-csp-compatible-template-with-self-hosted-static-files)

This PR also includes CI for testing and auto-update from`@openactive/dataset-site-template`.

### CODE REVIEW

File: examples/openactive-csp/index.php [hunk 1]
```diff
@@ -0,0 +1,45 @@
...
+
+require __DIR__ . "/../../vendor/autoload.php";
+
+use OpenActive\DatasetSiteTemplate\FeedType;
+use OpenActive\DatasetSiteTemplate\TemplateRenderer;
+
+// Get JSON-LD data
+
+// Create new dataset
+$settings = array(
```
Comment (nathansalter) [inline examples/openactive-csp/index.php:11]: You should use `[]` instead of `array()`

File: src/TemplateRenderer.php [hunk 2]
```diff
@@ -147,34 +152,59 @@ public function renderSimpleDatasetSiteFromDataDownloads(
             "datePublished" => $settings["dateFirstPublished"],
         ]);
 
+        if (isset($settings["openBookingAPIBaseUrl"])) {
+            $dataset->setAccessService(new WebAPI([
+                "name" => "Open Booking API",
+                "description" => "An API that allows for seamless booking experiences to be created for ".
+                    strtolower($dataFeedHumanisedList)." available from ".
```
Comment (nathansalter) [inline src/TemplateRenderer.php:159]: If this is a list, running `strtolower` will print out `array`. Use `array_map('strtolower', $dataFeedHumanisedList)` instead.

File: src/TemplateRenderer.php [hunk 3]
```diff
@@ -147,34 +152,59 @@ public function renderSimpleDatasetSiteFromDataDownloads(
...
         if (isset($settings["accessService"])) {
             $dataset->setAccessService($settings["accessService"]);
         }
 
         // Only set BookingService if valid platformName provided
-        if(
+        if((
             array_key_exists("platformName", $settings) &&
             is_string($settings["platformName"]) &&
             strlen($settings["platformName"]) > 0
```
Comment (nathansalter) [inline src/TemplateRenderer.php:182]: You can replace all three of these checks with `!empty($settings["platformName"])`

File: src/TemplateRenderer.php [hunk 4]
```diff
@@ -147,34 +152,59 @@ public function renderSimpleDatasetSiteFromDataDownloads(
...
-        ) {
-            $dataset->setBookingService(new BookingService([
-                "name" => $settings["platformName"],
-                "url" => $settings["platformUrl"],
-                "softwareVersion" => $settings["platformSoftwareVersion"],
-            ]));
+        ) || (
+            array_key_exists("testSuiteCertificateUrl", $settings) &&
+            is_string($settings["testSuiteCertificateUrl"]) &&
+            strlen($settings["testSuiteCertificateUrl"]) > 0
```
Comment (nathansalter) [inline src/TemplateRenderer.php:186]: Same comment as above

File: src/TemplateRenderer.php [hunk 5]
```diff
@@ -147,34 +152,59 @@ public function renderSimpleDatasetSiteFromDataDownloads(
...
+        )) {
+            $dataset->setBookingService(new BookingService(array_filter([
+                "name" => isset($settings["platformName"]) ? $settings["platformName"] : null,
```
Comment (nathansalter) [inline src/TemplateRenderer.php:189]: If we set the minimum version to PHP 7.0, we could use `$settings["platformName"] ?? null` :shrug:

File: src/datasetsite-csp.mustache [hunk 6]
```diff
@@ -0,0 +1,205 @@
...
+    
+    <!--
+      This stylesheet href must reference a self-hosted 'datasetsite.styles.v1.0.16.css' file in
+      the same directory as the other static assets sourced from the following archive:
+      https://unpkg.com/@openactive/dataset-site-template@1.0.16/dist/datasetsite-csp.static.zip
+    -->
+    <link rel="stylesheet" href="{{{staticAssetsPathUrl}}}/datasetsite.styles.v1.0.16.css" crossorigin="anonymous">
+
+</head>
+<body id="booking" background="{{backgroundImage.url}}">
```
Comment (nathansalter) [inline src/datasetsite-csp.mustache:57]: If this URL has any special chars in it then it'll get HTML encoded here, might be better to use triple brackets as above

File: tests/Unit/TemplateRendererTest.php [hunk 7]
```diff
@@ -27,17 +27,43 @@ public function testTemplateRendererInstance($renderer, $data, $supportedFeedTyp
...
      */
     public function testRenderString($renderer, $data, $supportedFeedTypes)
     {
-        $this->assertInternalType(
-            "string",
-            $renderer->renderSimpleDatasetSite($data, $supportedFeedTypes)
-        );
+        // assertInternalType has been deprecated by the newer PHPUnit, however it is still required here for when the tests run in PHP 5.6 / PHPUnit 5.7
+        $outputHtml = $renderer->renderSimpleDatasetSite($data, $supportedFeedTypes);
+        if (method_exists($this,'assertIsString')) {
```
Comment (nathansalter) [inline tests/Unit/TemplateRendererTest.php:39]: You shouldn't put logic in tests, maybe split up the data provider into the `assertIsString` and `assertInternalType` tests instead

File: tests/Unit/TemplateRendererTest.php [hunk 8]
```diff
@@ -27,17 +27,43 @@ public function testTemplateRendererInstance($renderer, $data, $supportedFeedTyp
...
+     * Test that the template renderer renders a string for the CSP template.
+     *
+     * @dataProvider templateRendererProvider
+     * @return void
+     */
+    public function testRenderStringCSP($renderer, $data, $supportedFeedTypes)
+    {
+        // assertInternalType has been deprecated by the newer PHPUnit, however it is still required here for when the tests run in PHP 5.6 / PHPUnit 5.7
+        $outputHtml = $renderer->renderSimpleDatasetSite($data, $supportedFeedTypes, "/example/asset/path");
+        if (method_exists($this,'assertIsString')) {
```
Comment (nathansalter) [inline tests/Unit/TemplateRendererTest.php:59]: Same comment as above

**Review by nathansalter — CHANGES_REQUESTED** [review]
This is pretty good! Just a few small things, some things which can be simplified a lot :clap:

**Comment by drinkynet** [comment]
Both of the png images in the static assets zip file seem to be broken. Neither previewer no any browsers can display them. Firefox says "cannot be displayed because it contains errors", the others just show either a white box on a black background or the broken image icon.

File: src/TemplateRenderer.php (hunk 2 above)
Comment (nickevansuk) [reply src/TemplateRenderer.php:159]: Confusingly `$dataFeedHumanisedList` appears to be a string. Have renamed to `$dataFeedListHumanisedString` to clarify.

File: src/TemplateRenderer.php (hunk 3 above)
Comment (nickevansuk) [reply src/TemplateRenderer.php:182]: 🙌 nice!

File: src/TemplateRenderer.php (hunk 5 above)
Comment (nickevansuk) [reply src/TemplateRenderer.php:189]: null coalescing is excellent! I remember when this came to Node.js not so long ago.

**Review by nathansalter — APPROVED** [review]

**Comment by nickevansuk** [comment]
Thanks @drinkynet - that’s such a great find with the static assets - it’s actually a bug in the `dataset-site-template`, which I've just fixed separately and will let the auto PR in CI update after this is merged

**Pull request merged**